utils/vptr/tests/basic.cc

```diff
@@ -0,0 +1,212 @@
+/***************************************************************************
+ *
+ *  Copyright (C) 2016 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's ComputeCpp SDK
+ *
+ *   basic.cc
+ *
+ *  Description:
+ *   Basic tests of the pointer mapper utility header
+ *
+ **************************************************************************/
+
+#include "gtest/gtest.h"
+
+#include <CL/sycl.hpp>
+#include <iostream>
+
+#include "virtual_ptr.hpp"
+#include "pointer_alias.hpp"
+
+using sycl_acc_target = cl::sycl::access::target;
+const sycl_acc_target sycl_acc_host = sycl_acc_target::host_buffer;
+
+using sycl_acc_mode = cl::sycl::access::mode;
+const sycl_acc_mode sycl_acc_rw = sycl_acc_mode::read_write;
+
+using namespace codeplay;
+
+using buffer_t = PointerMapper<>::buffer_t;
+
+TEST(pointer_mapper, basic_test) {
+  PointerMapper<> pMap;
+  {
+    ASSERT_EQ(pMap.count(), 0u);
+    void *myPtr = SYCLmalloc(100 * sizeof(float), pMap);
+    ASSERT_NE(myPtr, nullptr);
+
+    ASSERT_FALSE(PointerMapper<>::is_nullptr(myPtr));
+
+    ASSERT_TRUE(PointerMapper<>::is_nullptr(nullptr));
+
+    ASSERT_EQ(pMap.count(), 1u);
+
+    buffer_t b = pMap.get_buffer(myPtr);
+
+    cl::sycl::queue q;
+    q.submit([&b](cl::sycl::handler &h) {
+      auto accB = b.get_access<sycl_acc_rw>(h);
+      h.single_task<class foo1>([=]() { accB[0] = 1.0f; });
```

Comment:
There is an implicit conversion here, `1.0f` is being converted to `int8_t` the type of the `buffer`.